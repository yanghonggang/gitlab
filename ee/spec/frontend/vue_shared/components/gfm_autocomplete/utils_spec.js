import {
  GfmAutocompleteType,
  tributeConfig,
} from 'ee/vue_shared/components/gfm_autocomplete/utils';

describe('ee gfm_autocomplete/utils', () => {
  describe('epics config', () => {
    const epicsConfig = tributeConfig[GfmAutocompleteType.Epics].config;
    const epic = {
      id: null,
      iid: 123456,
      title: "Epic title <script>alert('hi')</script>",
    };

    it('uses & as the trigger', () => {
      expect(epicsConfig.trigger).toBe('&');
    });

    it('inserts the iid on autocomplete selection', () => {
      expect(epicsConfig.fillAttr).toBe('iid');
    });

    it('searches using both the iid and title', () => {
      expect(epicsConfig.lookup(epic)).toBe(`${epic.iid}${epic.title}`);
    });

    it('shows the iid and title in the menu item', () => {
      expect(epicsConfig.menuItemTemplate({ original: epic })).toMatchSnapshot();
    });
  });
});
