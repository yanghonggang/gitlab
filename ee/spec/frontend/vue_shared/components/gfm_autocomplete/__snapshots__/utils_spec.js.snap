// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ee gfm_autocomplete/utils epics config shows the iid and title in the menu item 1`] = `"<small>123456</small> Epic title &lt;script&gt;alert(&#39;hi&#39;)&lt;/script&gt;"`;
